exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -190,10 +193,13 @@ public QProfiles(List<ProfileInfo> runningQueries, List<ProfileInfo> finishedQue
     public List<String> getErrors() { return errors; }
   }
 
+  //max Param to cap listing of profiles 
+  private static final String MAX_QPROFILES_PARAM = "max";
```

Comment:
Lucky. There is a WorkManager instance available. We're able to load the default from `drilloverride.conf` and also have the user pass it via the URL.